File: environments_harbor/gradle_build_cache_configuration_medium/environment/java-app/src/main/java/com/example/Calculator.java
package com.example;

/**
 * A simple calculator utility class for basic arithmetic operations.
 * This class provides methods for addition, subtraction, multiplication, and division.
 */
public class Calculator {

    /**
     * Adds two integers and returns the result.
     * @param a first integer
     * @param b second integer
     * @return sum of a and b
     */
    public int add(int a, int b) {
        return a + b;
    }

    /**
     * Subtracts the second integer from the first and returns the result.
     * @param a first integer
     * @param b second integer
     * @return difference of a and b
     */
    public int subtract(int a, int b) {
        return a - b;
    }

    /**
     * Multiplies two integers and returns the result.
     * @param a first integer
     * @param b second integer
     * @return product of a and b
     */
    public int multiply(int a, int b) {
        return a * b;
    }

    /**
     * Divides the first integer by the second and returns the result.
     * @param a dividend
     * @param b divisor
     * @return quotient of a divided by b
     * @throws ArithmeticException if b is zero
     */
    public double divide(int a, int b) {
        if (b == 0) {
            throw new ArithmeticException("Division by zero");
        }
        return (double) a / b;
    }
}

File: environments_harbor/coverage_llvm_source_based_medium/environment/workspace/calculator/calculator.c
/*
 * Calculator Library
 * A simple arithmetic operations library
 * Legacy code - in production since 2015
 */

#include <stdio.h>

/* Addition function
 * Returns the sum of two integers
 */
int add(int a, int b) {
    return a + b;
}

/* Subtraction function
 * Returns the difference of two integers
 */
int subtract(int a, int b) {
    return a - b;
}

/* Multiplication function
 * Returns the product of two integers
 */
int multiply(int a, int b) {
    return a * b;
}

/* Division function
 * Returns the quotient of two integers
 * Returns 0 if divisor is zero to avoid crash
 */
int divide(int a, int b) {
    if (b == 0) {
        fprintf(stderr, "Error: Division by zero\n");
        return 0;
    }
    return a / b;
}

/* Modulo function
 * Returns the remainder of integer division
 * Returns 0 if divisor is zero
 */
int modulo(int a, int b) {
    if (b == 0) {
        fprintf(stderr, "Error: Modulo by zero\n");
        return 0;
    }
    return a % b;
}